lib/Controller/ViewController.php

```diff
@@ -148,4 +149,18 @@ private function getView(string $view): string {
 				return $view;
 		}
 	}
+
+	/**
+	 * @NoCSRFRequired
+	 *
+	 * This function makes the colour dots work for mobile widgets
+	 *
+	 * Returns an SVG with size32x32
+	 */
+	public function getCalendarDotSvg(string $color) {
+		$svg = '<svg height="32" width="32" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg"><rect width="100%" height="100%" fill="' . urldecode($color) . '"/></svg>';
```

Comment:
Maybe some validation is better to make sure this is not abused? `calendar-color` *should* always be an hexadecimal color.

Reply:
good point

Reply:
FYI, to make sure I tested on a production database and only 0.004% of entries didn't match that. They were lacking the leading `#`, which probably meant a bugged 3rd-party client.

In any case I think the behavior should be to reset `$color` to a valid one, not sure about sending an exception like this.

Reply:
yes another good point, thanks for testing on a live instance. I changed the logic to return a svg in any case, using the default nextcloud blue as a fallback.